pex/bin/pex.py

```diff
@@ -525,6 +530,16 @@ def build_pex(args, options, resolver_option_builder):
       for interpreter in options.python or [None]
     ]
 
+  if options.interpreter_constraint:
+    # Overwrite the current interpreter as defined by sys.executable.
+    # NB: options.python and interpreter constraints cannot be used together, so this will not
+    # affect usages of the interpreter(s) specified by the "--python" command line flag.
+    constraints = options.interpreter_constraint
+    validate_constraints(constraints)
+    rc_variables = Variables.from_rc(build_time_rc_dir=os.path.dirname(options.pex_name))
```

Comment:
its still puzzling to me why we need to look at a `.pexrc` relative to the output pex name for `PPP` configuration at build time.. can you clarify that with a comment here?

if this is purely for tests, we should probably re-evaluate.

Reply:
The use case is when the output pex is not in cwd, i.e. `-o dir/output.pex`. In that case, we should be supporting the case where the .pexrc is in `dir` as per a precedent set in the tests, however I see this being practical beyond testing situations. 

Previously, the check was for os.path.dirname(sys.argv[0]) to find this .pexrc in the same dir as the output pex at runtime, which works fine because sys.argv[0] is the pex itself, however, at build time, sys.argv[0] is the pex cli binary, which can be different than the dir of the output pex i.e. `pex requests -o dir/output.pex`. In this case, where I have .pexrc in `dir`, special accommodations are needed to make sure that the pex build routine can find that pexrc at build time. 

Like you say, this precedent is set largely by the tests, however having a pexrc in the same location as the pex itself seems like a reasonable case to cover to me.